### ISSUE TITLE
girder DB item refresh sometimes messes up UpSet

### ISSUE THREAD

**Issue by curtislisle** [ronichoudhury/nci-nanoparticles-vm#8]
sometimes after a project has been closed and re-opened, Upset remembers the defined sets, but the values are all blank.  In this case there is only one resulting row in UpSet with all variables = 0.    Even if the mapping is changed, UpSet doesn't correctly reinitialize.  It seems to often happen when the Girder DB item takes a while to refresh.  This is with the version from 5/30.  Should I create a new instance with master of candela to see if the problem is resolved in origin/master?

**Comment by waxlamp** [comment]
You could try that but I'm not sure if it will help. More likely we will have to debug to see what's going on. Let me know what you'd like to do.

**Comment by curtislisle** [comment]
Where in ResLab code is the repo of the project?  I am wondering about adding a delay before restoring the visualization to give the girder_db_item more time to initialize? 

I will package up the VM as is and forward a copy to NCI for experimentation since it works most of the time.  This VM could be available to Kitware as well, so we can see if we can replicate the issue. We may be seeing this because I’m pushing the biggest data through the pipeline currently and we’ve induced a timing race conditions. 

I’ll let you know when I have it ready.  In the meantime, it has been exciting to see a flurry of action around the extensions of Res Lab capabilities!

**Comment by waxlamp** [comment]
@alex-r-bigelow is this bug related to the one you were/are working on today?

**Comment by alex-r-bigelow** [comment]
This sounds similar to the issue that Curt mentioned that I couldn't seem to reproduce—not sure if it's exactly the same, but there's a chance the [latest fix](https://github.com/Kitware/candela/commit/48358b7992d194b446a835f7f877671e5eb00f36) has dealt with it.

@curtislisle, about how big was the file that was causing problems (just to help me reproduce the issue)?

**Comment by curtislisle** [comment]
Yes, this email thread is one and the same issue that Alex may have already fixed.  The source file was 3200 rows by about 65 columns.  The source CSV file used to load into mongo is here:

https://github.com/ronichoudhury/nci-nanoparticles-vm/blob/master/files/nano-0516-filled.csv https://github.com/ronichoudhury/nci-nanoparticles-vm/blob/master/files/nano-0516-filled.csv

I am thinking the majority of any delay is coming from pulling these records through from mongo as a girder db item.  I was thinking that I would upload this file twice to mongo, then I’d have plenty or records :-) 

I could see us wanting to run pretty long databases through this girder DB item path eventually.  It is a very powerful concept!
